kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/BigQuerySinkTask.java

```diff
@@ -247,7 +249,8 @@ public void put(Collection<SinkRecord> records) {
           TableWriterBuilder tableWriterBuilder;
           if (config.getList(BigQuerySinkConfig.ENABLE_BATCH_CONFIG).contains(record.topic())) {
             String topic = record.topic();
-            String gcsBlobName = topic + "_" + uuid + "_" + Instant.now().toEpochMilli();
+            long offset = record.kafkaOffset();
+            String gcsBlobName = topic + "_" + uuid + "_" + Instant.now().toEpochMilli()+"_"+records.size()+"_"+offset;
```

Comment:
Would having a test case for validating that parallel puts create different files with the right offset help?

Reply:
gcp side 
- you can use gcs versioning and how many times write gcs name.

bigquery sink 
- it's concept is used s3 sink. i convert to bigquery sink. 
  it is using first offset of records. and it is not corrupted because kafka messages unique.

if you need more comment, connect me.